ui/src/preview.tsx

```diff
@@ -0,0 +1,117 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Model, S } from 'h2o-wave'
+import React from 'react'
+import { stylesheet } from 'typestyle'
+import { Component, XComponents } from './form'
+import { cards } from './layout'
+import { clas, cssVar, cssVarValue } from './theme'
+import { bond, wave } from './ui'
+import * as Fluent from '@fluentui/react'
+
+const css = stylesheet({
+  card: {
+    borderRadius: 4,
+  },
+  img: {
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'space-between',
+    height: '100%',
+    boxSizing: 'border-box',
+    backgroundSize: 'cover',
+    backgroundRepeat: 'no-repeat',
+    backgroundPosition: 'center',
+    borderRadius: 4,
+  },
+  header: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    marginBottom: 8,
+  },
+  title: {
+    color: cssVar('$black') ,
+  },
+  caption: {
+    color: cssVar('$neutralPrimary') ,
+  },
+  content: {
+    padding: 24,
+    paddingBottom: 100,
+  },
+  items: {
+    $nest: {
+      // Target mini button's icon and label
+      '& .ms-Icon, & .ms-Button-label': {
+        color: cssVar('$black'),
+      }
+    }
+  }
+})
+
+
+/** Create a preview card displaying an image with shadow overlay, title, social icons, caption, and button. */
+interface State {
+  /** An identifying name for this card. Makes the card clickable if label is not provided, similar to a button. */
+  name: S
+  /** The card’s image. */
+  image: S
+  /** The card's title */
+  title?: S
+  /** Component's displayed at the top-right corner */
+  items?: Component[]
+  /** The card's caption, displayed bellow the title. */
+  caption?: S
+  /** Label of a button rendered at the bottom of the card. If specified, whole card is not clickable anymore.. */
+  label?: S
+}
+
+export const View = bond(({ name, state, changed }: Model<State>) => {
+  const render = () => {
+    const { image, title, name: stateName, label, caption, items } = state,
+    onClick = () => {
+      if (stateName.startsWith('#')) {
+        window.location.hash = stateName.substr(1)
+        return
+      }
+      wave.args[stateName] = stateName
+      wave.push()
+    },
+    getGradient = () => {
+      const { r, g, b } = Fluent.getColorFromString(cssVarValue('$card')) || { r: 255, g: 255, b: 255 }
+      return `linear-gradient(rgba(${r}, ${g}, ${b}, .8) 50%, rgba(${r}, ${g}, ${b}, 0) 100%)`
+    }
+
+    return (
+      <div data-test={name} className={css.card}>
+        <div className={css.img} style={{ backgroundImage: `url('${image}')` }}>
```

Comment:
Also rename css.img to css.card.
```suggestion
      <div data-test={name} className={css.img} style={{ backgroundImage: `url('${image}')` }}>
```